Sources/swift-algorand-sdk/v2/client/indexer/LookUpApplicationLogsById.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jesulonimi Akingbesote on 19/01/2022.
+//
+
+import Foundation
+import Alamofire
+
+public class LookUpApplicationLogsById{
+    var client:IndexerClient
+    var rawTransaction:[Int8]?
+    var applicationId:Int64
+    var queryItems:[String:String]=[:]
+    
+    init(client:IndexerClient,applicationId:Int64) {
+        self.client=client
+        self.applicationId = applicationId
+    }
+    
+    public func execute( callback: @escaping (_:Response<ApplicationLogResponse>) ->Void){
+        
+        let headers:HTTPHeaders=[client.apiKey:client.token]
+        var request=AF.request(getRequestString(parameter: self.applicationId),method: .get, parameters: nil, headers: headers,requestModifier: { $0.timeoutInterval = 120 })
+        request.validate()
+        var customResponse:Response<ApplicationLogResponse>=Response()
+  request.responseDecodable(of: ApplicationLogResponse.self){  (response) in
+
+    if(response.error != nil){
+        customResponse.setIsSuccessful(value:false)
+        var errorDescription=String(data:response.data ?? Data(response.error!.errorDescription!.utf8),encoding: .utf8)
+        customResponse.setErrorDescription(errorDescription:errorDescription!)
+        callback(customResponse)
+        if(response.data != nil){
+            if let message = String(data: response.data!,encoding: .utf8){
+                var errorDic = try! JSONSerialization.jsonObject(with: message.data, options: []) as? [String: Any]
+                customResponse.errorMessage = errorDic!["message"] as! String
+            }
+        }
+        return
+    }
+                    let data=response.value
+                    var applicationLogResponse:ApplicationLogResponse=data!
+                    customResponse.setData(data:applicationLogResponse)
+                    customResponse.setIsSuccessful(value:true)
+                    callback(customResponse)
+
+    }
+    }
+    
+    
+  
+    public func limit(limit:Int64) ->LookUpApplicationLogsById{
+        self.queryItems["limit"] = "\(limit)"
+        return self;
+        }
+    public func maxRound(maxRound:Int64) ->LookUpApplicationLogsById{
+        self.queryItems["maxRound"] = "\(maxRound)"
+        return self;
+        }
+    
+    public func minRound(minRound:Int64) ->LookUpApplicationLogsById{
+        self.queryItems["minRound"] = "\(minRound)"
+        return self;
+        }
+    
+    public func next(next:String) ->LookUpApplicationLogsById{
+        self.queryItems["next"] = next
+        return self;
+        }
+
+    
+    public func senderAddress(senderAddress:Address) ->LookUpApplicationLogsById{
+        self.queryItems["senderAddress"] = senderAddress.description
+        return self;
+        }
```

Comment:
I don't get this kind of syntax. 
It's really dangerous, as it force users to manipulate the request itself **after** it has been created.

We should simply use the constructor (`init()`) with all parameters needed, it's way more easier and safe don't you think?

Reply:
I mean, it's not something a Swift dev would expect and normally use for this kind of scenario: the HTTP request should be created at once and executed immediately. 
It's not like - example - a UILabel that you have to change multiple times in the lifecycle of the app.

Reply:
Thanks for this, i guess i'll create another PR that fixes this since there are other files that have this problem